### PR TITLE
Move AdvSimd check from PAL to VM

### PR DESCRIPTION
**Pull request by kunalspathak**
As per discussion in https://github.com/dotnet/runtime/pull/38060#discussion_r444069585 , making following changes:
- Remove redundant for `Set64BitInstructionSetVariants()` and `CPUCompileFlags.EnsureValidInstructionSetSupport()` from PAL
- Reverted part of the changes I did in https://github.com/dotnet/runtime/pull/38060 and move the check of `AdvSimd` presence in case of `TARGET_UNIX` from PAL to `codeman.cpp` .

### CODE REVIEW

File: src/coreclr/src/vm/codeman.cpp [hunk 1]
```diff
@@ -1483,6 +1483,15 @@ void EEJitManager::SetCpuInfo()
     }
 #if defined(TARGET_UNIX)
     PAL_GetJitCpuCapabilityFlags(&CPUCompileFlags);
+
+    // For HOST_ARM64, if OS has exposed mechanism to detect CPU capabilities, make sure it has AdvSimd capability.
+    // For other cases i.e. if !HOST_ARM64 but TARGET_ARM64 or HOST_ARM64 but OS doesn't expose way to detect
+    // CPU capabilities, we always enable AdvSimd flags by default.
+    //
+    if (!CPUCompileFlags.IsSet(InstructionSet_AdvSimd))
```
Comment (tannergooding) [inline src/coreclr/src/vm/codeman.cpp:1491]: Can we go ahead and add the same for SSE/SSE2 for x86?

Comment (tannergooding) [reply src/coreclr/src/vm/codeman.cpp:1500]: Thanks!

**Comment by kunalspathak** [comment]
@janvorli , @davidwrighton , @BruceForstall

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Review by davidwrighton — APPROVED** [review]

**Pull request merged**